### PR TITLE
[app_dart] Create service account authentication provider

### PR DESCRIPTION
**Pull request by CaseyHillers**
Refactor `luci-status` to use the existing auth methods

This is necessary to create `KeyHelper` in LuciStatus so we can update prod tasks in datastore via pubsub

### CODE REVIEW

File: app_dart/test/request_handling/authentication_test.dart [hunk 1]
```diff
@@ -60,67 +62,6 @@ void main() {
         );
       });
 
-      test('auth succeeds with authenticated cookie but unauthenticated header', () async {
```
Comment (CaseyHillers) [inline app_dart/test/request_handling/authentication_test.dart:63]: These tests should have been removed in the agent cleanup. They have equivalents else where

File: app_dart/lib/src/service/luci_build_service.dart [hunk 2]
```diff
@@ -456,6 +456,7 @@ class LuciBuildService {
     @required String builderName,
     String branch = 'master',
     String repo = 'flutter',
+    String encodedTaskKey,
```
Comment (keyonghan) [inline app_dart/lib/src/service/luci_build_service.dart:459]: Nit: doc for `encodedTaskKey`.

**Review by keyonghan — COMMENTED** [review]
LGTM with nit.
As before, can we verify the functionality with some traffic before landing?

**Comment by CaseyHillers** [comment]
Starting traffic test now with testchillers

File: app_dart/lib/src/service/luci_build_service.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_dart/lib/src/service/luci_build_service.dart:461]: Done

**Comment by CaseyHillers** [comment]
Traffic test showed authentication is still working as intended

**Review by keyonghan — APPROVED** [review]

**Pull request merged**